File: 0001_0100/0055/0055.py
import math

with open("input.txt", "r") as f, open("output.txt", "w") as q:
    x1, y1, x2, y2, r, s = map(int, f.read().split())
    k = ((x1 - x2) ** 2 + (y1 - y2) ** 2) ** 0.5 / 2
    a = 2 * math.acos((k / r) % 1)
    sn = 2 * math.pi * r * r if k >= r else r * r * (2 * math.pi - a + math.sin(a))
    q.write("YES" if sn > s else "NO")

File: 0001_0100/0055/0055.go
package main

import (
	"fmt"
	"math"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var x1, y1, x2, y2, r, s float64
	fmt.Fscan(f, &x1, &y1, &x2, &y2, &r, &s)
	k := math.Sqrt((x1-x2)*(x1-x2)+(y1-y2)*(y1-y2)) / 2
	a := 2 * math.Acos(k/r)
	sn := 2 * math.Pi * r * r
	if k < r {
		sn = r * r * (2*math.Pi - a + math.Sin(a))
	}
	ans := "NO"
	if sn > s {
		ans = "YES"
	}
	fmt.Fprint(q, ans)
}
